initial commit of configEditor

carrier92 opened the pull request:
remove of all comments from config.template.ini

carrier92 committed:
initial commit of configEditor

remove of all comments from config.template editor

carrier92 committed:
code optimization

Schrolli91 commented:
Was mir ehrlich gesagt gar nicht gefällt, ist die Tatsache, das nun die Kommentare in der config weg sind... Es darf niemand gezwungen werden das Script zu nutzen...

Da die Kommentare aber nicht gefiltert werden ist das Script aktuell eher nicht lauffähig

carrier92 commented:
Okay mal schauen was ich da machen kann

Schrolli91 commented:
Ja im Prinzip, die vorhandene config so lassen wie sie aktuell ist, und eben beim Einlesen irgendwie die Kommentare rausfiltern 

Und wieso nutzt du nicht die `parse_ini_file` Methode die PHP mitbringt, wie in deinem ersten Code? Hat das einen bestimmten Grund, das du das parsing selbst implementiert hast?

carrier92 commented:
Der Grund warum ich nicht die parse_ini_file funktion verwendet habe ist das die config.ini nicht 100% PHP kompatibel ist, Texte mit Sonderzeichen müssten in doppelten Anführungszeichen geschrieben sein.
Du verwendest zwar die erweiterung .ini aber letztendlich ist es eine Python spezifische config (config_parser) Datei, die sich doch zu einer ini unterscheidet (Sonderzeichen, umbrüche in den Werten ...).

Machbar ist das schon, allerdings müssten dann die Kommentare mit einem semikolon markiert werden damit das ganze auch PHP7 kompatible bleibt.
Die Komentare beim einlesen zu ignorieren ist nicht das Problem.
Da ich aber beim schreiben die ganze Datei neu aufbaue wären die Kommentare danach weg.
Also wer sich einmal für das bearbeiten der config.ini mit meinem script entschieden hat bekommt die Kommentare nicht mehr in die ini Datei.

Schrolli91 commented:
Klingt logisch mit der ini kompatibelität... 

Ist mir klar das die Kommentare nach dem schreiben weg sind... Wer den Editor nutzt braucht die ja auch net.. Hätte sie aber gern in der original Datei drin zwecks den Leuten die sie händisch bearbeiten, wenn du verstehst was ich meine

carrier92 commented:
Vieleicht fällt mir noch was ein damit doch beides geht :)

Schrolli91 commented:
Mann könnte beim lesen der INI die Kommentare überspringen, und beim schreiben, dann die description Texte wieder als Kommentar schreiben... Dann bleiben sie in der INI File ;-)

carrier92 commented:
Ja so eine Lösung hab ich gerade in Arbeit :)

Schrolli91 commented:
Top...
Da bin ich ja mal gespannt... Ich denke wenn gegeben bleibt, dass man es auch ohne das Script sauber konfigurieren kann ist das eine tolle Sache...

carrier92 committed:
write comments back in to config.ini

Editor read description text from config_description.txt into
config.ini.
Form layout changed.
Form checks for illegal character.

flothi commented:
In #229 wurde u.a. vorgeschlagen, die Configeinstellungen in eine (zu erstellende) sqlite-DB auszulagern - eine gute Idee, wenn denn gleichzeitig alle "lebenswichtigen" Dinge (Pfade zu rtl_fm, Speicherort der DB) in der ini-Datei verbleiben - würde evtl den Entwicklungsaufwand reduzieren.

carrier92 committed:
Fehler in den Beschreibungen korrigiert

Schrolli91 commented:
Also dann der Plan, eine sqLite DB aufsetzen, welche auch vom SQL Plugin genutzt werden kann, aber Primär mal für config genutzt wird? Finde das klingt durchaus nicht verkehrt... Möglichkeit zum In- und Export müssten dann aber geschaffen werden

carrier92 commented:
Hallo,

wenn wir auf sqlite gehen hat sich diese pullrequest ja erledigt.
könnt ihr den löschen?

Ich hab mir mal Gedanken zu eine sqlite Datenbank gemacht.
Ein ersten versuch inklusive ein script um die bestehende config.ini in die DB zu schreiben hab ich auch schon.

flothi commented:
Bis sqlite wirklich drin ist würde ich den evtl doch stehen lassen - löschen kann man ihn dann immer noch.
